src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -103,8 +101,10 @@ namespace ripple {
         If the SusPay ledger entry specifies sfCancelAfter, the
         transaction will fail if sfCancelAfter <= parentCloseTime.
 
-        NOTE: It must always be possible to verify the condition
-              without retrieving the SusPay ledger entry.
+        NOTE: The reason the condition must be specified again
+              is because it must always be possible to verify
+              the condition without retrieving the SusPay
+              ledger entry.
```

Comment:
If you're going to do that, it's pointless to do anything in `SusPayFinish::preflight`. Just do the work in `SusPayFinish::doApply`. But that's what we want to avoid.

As designed, we can perform operations such as parsing a cryptocondition and doing cryptographic validation in `SusPayFinish::preflight` and in `SusPayFinish::doApply` all we have a simple comparison of two `Slice` objects.